src/glob.rs

```diff
@@ -303,16 +303,77 @@ impl PathOrPatternSet {
     ))
   }
 
-  pub fn from_relative_path_or_patterns(
+  /// Builds the set and ensures no negations.
+  pub fn from_include_relative_path_or_patterns(
     base: &Path,
-    paths: &[String],
+    entries: &[String],
   ) -> Result<Self, anyhow::Error> {
-    Ok(Self(
-      paths
-        .iter()
-        .map(|p| PathOrPattern::from_relative(base, p))
-        .collect::<Result<Vec<_>, _>>()?,
-    ))
+    let mut result = Vec::with_capacity(entries.len());
+    for entry in entries {
+      let p = PathOrPattern::from_relative(base, entry)?;
+      // for now, just don't support this until someone wants
+      // it even though it should work well with the current code
+      if p.is_negated() {
+        bail!("Negated entry '{}' not supported in \"include\". Move to \"exclude\" instead.", entry)
+      }
```

Comment:
We've never supported negations in "include" so it probably makes sense to keep not supporting them when we have "exclude".

Reply:
Actually, this is nice when someone provides globs via flags. I'll revert this.